data/notebooks/Lab/pu_lockinLockout_logic.ipynb

```diff
@@ -0,0 +1,51 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "af53f37a",
+   "metadata": {},
+   "source": [
+    "A Geometry is passed for lockin or lockout operations, we need to:\n",
+    "* Transform that geometry if it is a shapefile.\n",
+    "* Validate that geometry (valid polygon in projection 4326)\n",
+    "\n",
+    "```sql\n",
+    "UPDATE scenarios_pu_data\n",
+    "SET (lockin_status) = <2 or 3>\n",
+    "where id in (select id from scenarios_pu_data spd \n",
+    "inner join planning_units_geom pug \n",
+    "on spd.pu_geom_id = pug.id and spd.scenario_id ='<scenario_id>' and st_intersects(pug.the_geom, ST_GeomFromGeoJSON('')))\n",
```

Comment:
I am aware that this is still WIP but just a note - to speed things a little, we need to start adding indexes at least on ID columns that we often narrow things down.